src/Parser.php

```diff
@@ -569,7 +570,7 @@ protected function popTags()
     /**
      * Checks the tags fit the required format.
      *
-     * @param string[] $tags
+     * @param array<array-key, string> $tags
```

Comment:
Fairly sure this should be a list?

Reply:
this particular method does not care about keys, but we indeed always pass a list in practice in our own code.

I did not made the parameter stricter in my previous PR because this is a protected method. A child class could be passing a non-list value, and the implementation supports it without any change (as it ignores keys when looping anyway).

Reply:
I changed it back to `array<array-key` ... Please reopen if there is any disagreement.